xla/pjrt/cpu/cpu_client_test.cc

```diff
@@ -446,6 +446,50 @@ TEST(TfrtCpuClientTest, ForwardUserDataToFfiHandler) {
       *result_literal));
 }
 
+static absl::Status MemsetFromAttr(
+    float attr, ffi::Result<ffi::BufferR1<PrimitiveType::F32>> result) {
+  for (size_t i = 0; i < result->element_count(); ++i) {
+    result->typed_data()[i] = attr;
+  }
+  return absl::OkStatus();
+}
+
+XLA_FFI_DEFINE_HANDLER(kMemsetFromAttr, MemsetFromAttr,
+                       ffi::Ffi::Bind()
+                           .Attr<float>("attr")
+                           .Ret<ffi::BufferR1<PrimitiveType::F32>>());
+
+XLA_FFI_REGISTER_HANDLER(ffi::GetXlaFfiApi(), "MemsetFromAttr", "HOST",
+                         kMemsetFromAttr);
+
+TEST(StreamExecutorGpuClientTest, PassAttrToFfiHandler) {
+  static constexpr char const* kProgram = R"(
+    HloModule ffi_handler
+    ENTRY main {
+      ROOT %custom-call = f32[4] custom-call(),
+          custom_call_target="MemsetFromAttr",
+          api_version=API_VERSION_TYPED_FFI,
+          // Takes an encoded version of {"opaque": "{attr = 3.0 : f32}"}
```

Comment:
Why such an ugly encoded string? Because of bytes type? Isn’t it ascii string? I’m worried that it makes HLO unreadable for humans.

Reply:
Yes. That's your suggestion :)
Changing bytes to string will work for most cases. But as you said, if you want to embed a serialized proto in that field, that field becomes unreadable as well. 

The design decision would be just not putting encoded bytes in any proto.

Reply:
Ok, but we are all the way in the proto land, maybe we can to

```
oneof {
 bytes opaque = 1;
 string attributes = 2;
}
```

`opaque` for legacy custom calls and `attrs` for FFI

Reply:
Proto thing resolved offline.